src/bson.c

```diff
@@ -862,8 +862,8 @@ void object_to_bson(zval *object, php_phongo_bson_flags_t flags, const char *key
 			tmp_ht = HASH_OF(obj_data);
 #endif
 
-			if (tmp_ht) {
-				ZEND_HASH_APPLY_COUNT(tmp_ht)++;
+			if (tmp_ht && ZEND_HASH_APPLY_PROTECTION(tmp_ht)) {
+				ZEND_HASH_INC_APPLY_COUNT(tmp_ht);
```

Comment:
This looks scary - I would check with @dstogov whether this is correct. What happens when a "static" array has recursive loops now?